pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,353 @@
+    data store for a CIM repository.  The API provides functions to:
+
+    1. Manage CIM namespaces in the data repository including creation, deletion
+       and getting a list of the available namespaces.
+    2. Accessing the object store within the repository for the objects of the
+       following CIM types: (CIM classes, CIM instances, and CIM qualifier
+       decelarations) so that methods of the ObjStoreAPI can be used to access
+       objects by namespace so that CIM objects can be manupulated in the
+       repository by namespace.
+    """
+
+    @compatibleabstractproperty
+    def namespaces(self):
+        """
+        Readonly property that returns a list with the names of the
+        namespaces defined for this repository
+
+        Returns:
+            list of :term:`string` items containing the namespace names
+        """
+        pass
+
+    @abstractmethod
+    def validate_namespace(self, namespace):
+        """
+        Validate if the namespace parameter exists in the repository. If the
+        namespace does not exist a KeyError is raised.
+
+        Raises:
+            KeyError: If the namespace is not defined in the repository
+        """
+        pass
+
+    @abstractmethod
+    def add_namespace(self, namespace):
+        """
+        Add a CIM namespace to the repository.
+
+        The namespace must not yet exist in the mock repository.
+
+        The default connection namespace is automatically added to
+        the mock repository upon creation of this connection.
+
+          Parameters:
+            namespace (:term:`string`):
+              The name of the CIM namespace in the mock repository. Must not be
+              `None`. Leading and trailing slash characters are split off
+              from the provided string.
+
+          Raises:
+            ValueError: If the namespace argument is None or the
+             namespace already exists.
+        """
+        pass
+
+    @abstractmethod
+    def remove_namespace(self, namespace):
+        """
+        Remove a CIM namespace from the repository.
+
+        The namespace must exist in the repository and must be empty.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the mock repository. Must not be
+            `None`. Any leading and trailing slash characters are split off
+            from the provided string.
+
+        Raises:
+
+          ValueError: Namespace argument is None or the repository namespace
+            is not empty
+          KeyError:  The namespace does not exist in the mock repository.
+        """
+        pass
+
+    @abstractmethod
+    def get_class_repo(self, namespace):
+        """
+        Get the handle for the data CIM class object store for the namespace
+        provided.
+
+          Parameters:
+            namespace (:term:`string`):
+              The name of the CIM namespace in the mock repository. Must not be
+              `None`
```

Comment:
Add statements about leading and trailing slashes and about case insensitivity.

Reply:
Added to all uses of the namesspace parameter. DONE